(Flaky, intermittent) Build failure: edencommon (and therefore watchman) on arm64-darwin fails to build on Master

Fortunately, this now seems to just be a flaky test. I'll still file the bug, but it's not a serious issue.

This is the followup to the now-fixed https://github.com/NixOS/nixpkgs/issues/368501 . folly now builds correctly, but another intermediate dependency between it and watchman fails intermittently.

## Steps To Reproduce

Steps to reproduce the behavior:

1. build `edencommon` with e.g.

     nix build github:NixOS/nixpkgs/master#edencommon

(currently at commit 9e09ce2d47c7700b877becef99ce7e27f156)

2. Have bad luck so that the flaky addFromMultipleThreads test fails.


**Workaround:** Just build it again. I then had `watchman-2024.11.18` successfully build too. Woohoo.

## Build log


<details>

<summary>Build Log</summary>

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/kzkzmp2d92ppjdg4kbj1h18ik4h2fhq6-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
fixing cmake files...
cmake flags: -GNinja -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_LOCALEDIR=/nix/store/7py4cwdp6kidvhbafccbghmwjv320y8r-edencommon-2024.11.18.00/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/7py4cwdp6kidvhbafccbghmwjv320y8r-edencommon-2024.11.18.00/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/7py4cwdp6kidvhbafccbghmwjv320y8r-edencommon-2024.11.18.00/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/7py4cwdp6kidvhbafccbghmwjv320y8r-edencommon-2024.11.18.00/share/doc/edencommon -DCMAKE_INSTALL_INFODIR=/nix/store/7py4cwdp6kidvhbafccbghmwjv320y8r-edencommon-2024.11.18.00/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/7py4cwdp6kidvhbafccbghmwjv320y8r-edencommon-2024.11.18.00/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/mihlxhpj0z2hwjnjzqzr00miqpf0aa5s-edencommon-2024.11.18.00-dev/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/mihlxhpj0z2hwjnjzqzr00miqpf0aa5s-edencommon-2024.11.18.00-dev/include -DCMAKE_INSTALL_SBINDIR=/nix/store/7py4cwdp6kidvhbafccbghmwjv320y8r-edencommon-2024.11.18.00/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/7py4cwdp6kidvhbafccbghmwjv320y8r-edencommon-2024.11.18.00/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/7py4cwdp6kidvhbafccbghmwjv320y8r-edencommon-2024.11.18.00/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/wkadfwbhn9b4kdih88j7izzy20qw79bk-clang-wrapper-19.1.5/bin/strip -DCMAKE_RANLIB=/nix/store/wkadfwbhn9b4kdih88j7izzy20qw79bk-clang-wrapper-19.1.5/bin/ranlib -DCMAKE_AR=/nix/store/wkadfwbhn9b4kdih88j7izzy20qw79bk-clang-wrapper-19.1.5/bin/ar -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_INSTALL_PREFIX=/nix/store/7py4cwdp6kidvhbafccbghmwjv320y8r-edencommon-2024.11.18.00 -DBUILD_SHARED_LIBS:BOOL=TRUE -DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=TRUE -DINCLUDE_INSTALL_DIR:STRING=/nix/store/mihlxhpj0z2hwjnjzqzr00miqpf0aa5s-edencommon-2024.11.18.00-dev/include -DLIB_INSTALL_DIR:STRING=/nix/store/7py4cwdp6kidvhbafccbghmwjv320y8r-edencommon-2024.11.18.00/lib -DCMAKE_INSTALL_DIR:STRING=/nix/store/mihlxhpj0z2hwjnjzqzr00miqpf0aa5s-edencommon-2024.11.18.00-dev/lib/cmake/edencommon
-- The CXX compiler identification is Clang 19.1.5
-- The C compiler identification is Clang 19.1.5
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/wkadfwbhn9b4kdih88j7izzy20qw79bk-clang-wrapper-19.1.5/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/wkadfwbhn9b4kdih88j7izzy20qw79bk-clang-wrapper-19.1.5/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- setting C++ standard to C++17
-- Found Glog: /nix/store/0ap3kv0124w6dzmmlp299x4akqhs12q5-glog-0.6.0/lib/libglog.dylib
-- Found gflags from package config /nix/store/1lb1ip9h8n9ljdma08qfzims6hqy9g9z-gflags-2.2.2/lib/cmake/gflags/gflags-config.cmake
-- Found gflags as a dependency of glog::glog, include=/nix/store/1lb1ip9h8n9ljdma08qfzims6hqy9g9z-gflags-2.2.2/include, libs=gflags_shared
[33mCMake Warning (dev) at /nix/store/x8qyfphi04lhj621if89f8il33v9dnny-cmake-3.30.5/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:441 (message):
  The package name passed to `find_package_handle_standard_args` (LIBUNWIND)
  does not match the name of the calling package (LibUnwind).  This can lead
  to problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  build/fbcode_builder/CMake/FindLibUnwind.cmake:22 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  build/fbcode_builder/CMake/FindGlog.cmake:44 (find_package)
  CMakeLists.txt:53 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.
[0m
-- Could NOT find LIBUNWIND (missing: LIBUNWIND_LIBRARY) 
[33mCMake Warning (dev) at /nix/store/x8qyfphi04lhj621if89f8il33v9dnny-cmake-3.30.5/share/cmake-3.30/Modules/CMakeFindDependencyMacro.cmake:76 (find_package):
  Policy CMP0167 is not set: The FindBoost module is removed.  Run "cmake
  --help-policy CMP0167" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

Call Stack (most recent call first):
  /nix/store/dbx8agznm1cr54rk0r3kpdkvv6201w7b-folly-2024.11.18.00-dev/lib/cmake/folly/folly-config.cmake:59 (find_dependency)
  CMakeLists.txt:59 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.
[0m
-- Found Boost: /nix/store/gjbd4dvxahrhpazz0mf005jqfhcd650d-boost-1.86.0-dev/lib/cmake/Boost-1.86.0/BoostConfig.cmake (found suitable version "1.86.0", minimum required is "1.51.0") found components: context filesystem program_options regex system thread
-- Found folly: /nix/store/dbx8agznm1cr54rk0r3kpdkvv6201w7b-folly-2024.11.18.00-dev
-- Found fb303: /nix/store/80hg9689blbmmhl98pzqnx4snlszg1ra-fb303-2024.11.18.00-dev
-- Found wangle: /nix/store/slsm7pnr932kvhggl10f4w60q380n2sg-wangle-2024.11.18.00
-- Found ZLIB: /nix/store/5s3b3s1kc2i2y9j8zcm03nphg7lmcxrg-zlib-1.3.1/lib/libz.dylib (found version "1.3.1")
-- Found folly: /nix/store/dbx8agznm1cr54rk0r3kpdkvv6201w7b-folly-2024.11.18.00-dev
-- Found Sodium: /nix/store/klkv6v8gp7z2x9z6qifw4ad22108cfg4-libsodium-1.0.20/lib/libsodium.dylib
-- Found folly: /nix/store/dbx8agznm1cr54rk0r3kpdkvv6201w7b-folly-2024.11.18.00-dev
-- Found fizz: /nix/store/ikii8n0lvx85khjf3dbbcfwrr7bqs6zg-fizz-2024.11.18.00-dev
[33mCMake Warning (dev) at /nix/store/x8qyfphi04lhj621if89f8il33v9dnny-cmake-3.30.5/share/cmake-3.30/Modules/CMakeFindDependencyMacro.cmake:76 (find_package):
  Policy CMP0167 is not set: The FindBoost module is removed.  Run "cmake
  --help-policy CMP0167" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

Call Stack (most recent call first):
  /nix/store/jm2i04vvh256bns6vhf5x1dsg0vvx4fm-mvfst-2024.11.18.00-dev/lib/cmake/mvfst/mvfst-config.cmake:47 (find_dependency)
  /nix/store/1wligd654gjw7d7n2607rl7vb70xrqc0-fbthrift-2024.11.18.00/lib/cmake/fbthrift/FBThriftConfig.cmake:57 (find_package)
  CMakeLists.txt:68 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.
[0m
-- Found Boost: /nix/store/gjbd4dvxahrhpazz0mf005jqfhcd650d-boost-1.86.0-dev/lib/cmake/Boost-1.86.0/BoostConfig.cmake (found version "1.86.0") found components: iostreams system thread filesystem regex context
-- Found mvfst: /nix/store/jm2i04vvh256bns6vhf5x1dsg0vvx4fm-mvfst-2024.11.18.00-dev
-- Found FBThrift: /nix/store/1wligd654gjw7d7n2607rl7vb70xrqc0-fbthrift-2024.11.18.00
-- Found gmock via config, defines=, include=/nix/store/qis4x8vjbq81bbdlgpnss4c5fbkzzyb4-gtest-1.15.2-dev/include, libs=GTest::gmock_main;GTest::gmock;GTest::gtest
-- Configuring done (1.7s)
-- Generating done (0.0s)
[33mCMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_INSTALL_BINDIR
    CMAKE_INSTALL_DOCDIR
    CMAKE_INSTALL_INCLUDEDIR
    CMAKE_INSTALL_INFODIR
    CMAKE_INSTALL_LIBDIR
    CMAKE_INSTALL_LIBEXECDIR
    CMAKE_INSTALL_LOCALEDIR
    CMAKE_INSTALL_MANDIR
    CMAKE_INSTALL_OLDINCLUDEDIR
    CMAKE_INSTALL_SBINDIR
    CMAKE_POLICY_DEFAULT_CMP0025

[0m
-- Build files have been written to: /tmp/nix-build-edencommon-2024.11.18.00.drv-0/source/build
cmake: enabled parallel building
cmake: enabled parallel installing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j12
[0/2] Re-checking globbed directories...
[1/62] Building CXX object eden/common/utils/CMakeFiles/edencommon_utils.dir/Guid.cpp.o
[2/62] Building CXX object eden/common/utils/CMakeFiles/edencommon_utils.dir/IDGen.cpp.o
[3/62] Building CXX object eden/common/utils/CMakeFiles/edencommon_utils.dir/Memory.cpp.o
[4/62] Building CXX object eden/common/utils/CMakeFiles/edencommon_utils.dir/CoverageSet.cpp.o
[5/62] Building CXX object eden/common/utils/CMakeFiles/edencommon_utils.dir/FileDescriptor.cpp.o
[6/62] Building CXX object eden/common/utils/CMakeFiles/edencommon_utils.dir/Pipe.cpp.o
[7/62] Building CXX object eden/common/utils/CMakeFiles/edencommon_utils.dir/StringConv.cpp.o
[8/62] Building CXX object eden/common/utils/CMakeFiles/edencommon_utils.dir/ProcessInfo.cpp.o
[9/62] Building CXX object eden/common/utils/CMakeFiles/edencommon_utils.dir/Bug.cpp.o
[10/62] Building CXX object eden/common/utils/CMakeFiles/edencommon_utils.dir/ImmediateFuture.cpp.o
[11/62] Building CXX object eden/common/utils/CMakeFiles/edencommon_utils.dir/String.cpp.o
[12/62] Building CXX object eden/common/utils/CMakeFiles/edencommon_utils.dir/IoFuture.cpp.o
[13/62] Building CXX object eden/common/utils/CMakeFiles/edencommon_utils.dir/FileUtils.cpp.o
[14/62] Building CXX object eden/common/utils/CMakeFiles/edencommon_utils.dir/WinError.cpp.o
[15/62] Building CXX object eden/common/utils/CMakeFiles/edencommon_utils.dir/FutureUnixSocket.cpp.o
[16/62] Building CXX object eden/common/utils/CMakeFiles/edencommon_utils.dir/PathFuncs.cpp.o
[17/62] Building CXX object eden/common/utils/CMakeFiles/edencommon_utils.dir/SysctlUtil.cpp.o
[18/62] Building CXX object eden/common/utils/CMakeFiles/edencommon_utils.dir/Utf8.cpp.o
[19/62] Building CXX object eden/common/os/CMakeFiles/edencommon_os.dir/ProcessId.cpp.o
[20/62] Building CXX object eden/common/utils/CMakeFiles/edencommon_utils.dir/XAttr.cpp.o
[21/62] Building CXX object eden/common/utils/CMakeFiles/edencommon_utils.dir/TimeUtil.cpp.o
[22/62] Linking CXX shared library eden/common/os/libedencommon_os.dylib
[23/62] Building CXX object eden/common/testharness/CMakeFiles/edencommon_testharness.dir/TempFile.cpp.o
[24/62] Building CXX object eden/common/utils/CMakeFiles/edencommon_utils.dir/UserInfo.cpp.o
[25/62] Building CXX object eden/common/utils/CMakeFiles/edencommon_utils.dir/UnboundedQueueExecutor.cpp.o
[26/62] Building CXX object eden/common/telemetry/CMakeFiles/edencommon_telemetry.dir/SessionId.cpp.o
[27/62] Building CXX object eden/common/telemetry/CMakeFiles/edencommon_telemetry.dir/DynamicEvent.cpp.o
[28/62] Building CXX object eden/common/os/test/CMakeFiles/os_test.dir/ProcessIdTest.cpp.o
[29/62] Building CXX object eden/common/utils/CMakeFiles/edencommon_utils.dir/UnixSocket.cpp.o
/tmp/nix-build-edencommon-2024.11.18.00.drv-0/source/eden/common/utils/UnixSocket.cpp:427:52: warning: variable length arrays in C++ are a Clang extension [-Wvla-cxx-extension]
  427 |       sizeof(SendQueueEntry) + sizeof(struct iovec[entry->iovCount]);
      |                                                    ^~~~~~~~~~~~~~~
/tmp/nix-build-edencommon-2024.11.18.00.drv-0/source/eden/common/utils/UnixSocket.cpp:427:52: note: function parameter 'entry' with unknown value cannot be used in a constant expression
/tmp/nix-build-edencommon-2024.11.18.00.drv-0/source/eden/common/utils/UnixSocket.cpp:424:66: note: declared here
  424 | void UnixSocket::SendQueueDestructor::operator()(SendQueueEntry* entry) const {
      |                                                                  ^
/tmp/nix-build-edencommon-2024.11.18.00.drv-0/source/eden/common/utils/UnixSocket.cpp:427:52: warning: variable length arrays in C++ are a Clang extension [-Wvla-cxx-extension]
  427 |       sizeof(SendQueueEntry) + sizeof(struct iovec[entry->iovCount]);
      |                                                    ^~~~~~~~~~~~~~~
/tmp/nix-build-edencommon-2024.11.18.00.drv-0/source/eden/common/utils/UnixSocket.cpp:427:52: note: function parameter 'entry' with unknown value cannot be used in a constant expression
/tmp/nix-build-edencommon-2024.11.18.00.drv-0/source/eden/common/utils/UnixSocket.cpp:424:66: note: declared here
  424 | void UnixSocket::SendQueueDestructor::operator()(SendQueueEntry* entry) const {
      |                                                                  ^
/tmp/nix-build-edencommon-2024.11.18.00.drv-0/source/eden/common/utils/UnixSocket.cpp:445:52: warning: variable length arrays in C++ are a Clang extension [-Wvla-cxx-extension]
  445 |       sizeof(SendQueueEntry) + sizeof(struct iovec[iovecElements]);
      |                                                    ^~~~~~~~~~~~~
/tmp/nix-build-edencommon-2024.11.18.00.drv-0/source/eden/common/utils/UnixSocket.cpp:445:52: note: read of non-const variable 'iovecElements' is not allowed in a constant expression
/tmp/nix-build-edencommon-2024.11.18.00.drv-0/source/eden/common/utils/UnixSocket.cpp:441:10: note: declared here
  441 |   size_t iovecElements = 1;
      |          ^
/tmp/nix-build-edencommon-2024.11.18.00.drv-0/source/eden/common/utils/UnixSocket.cpp:445:52: warning: variable length arrays in C++ are a Clang extension [-Wvla-cxx-extension]
  445 |       sizeof(SendQueueEntry) + sizeof(struct iovec[iovecElements]);
      |                                                    ^~~~~~~~~~~~~
/tmp/nix-build-edencommon-2024.11.18.00.drv-0/source/eden/common/utils/UnixSocket.cpp:445:52: note: read of non-const variable 'iovecElements' is not allowed in a constant expression
/tmp/nix-build-edencommon-2024.11.18.00.drv-0/source/eden/common/utils/UnixSocket.cpp:441:10: note: declared here
  441 |   size_t iovecElements = 1;
      |          ^
4 warnings generated.
[30/62] Building CXX object eden/common/telemetry/CMakeFiles/edencommon_telemetry.dir/SessionInfo.cpp.o
[31/62] Building CXX object eden/common/telemetry/CMakeFiles/edencommon_telemetry.dir/RequestMetricsScope.cpp.o
[32/62] Building CXX object eden/common/telemetry/CMakeFiles/edencommon_telemetry.dir/StructuredLogger.cpp.o
[33/62] Building CXX object eden/common/utils/CMakeFiles/edencommon_utils.dir/ProcessInfoCache.cpp.o
[34/62] Building CXX object eden/common/telemetry/CMakeFiles/edencommon_telemetry.dir/ScubaStructuredLogger.cpp.o
[35/62] Building CXX object eden/common/utils/CMakeFiles/edencommon_utils.dir/SpawnedProcess.cpp.o
[36/62] Building CXX object eden/common/telemetry/CMakeFiles/edencommon_telemetry.dir/SubprocessScribeLogger.cpp.o
[37/62] Building CXX object eden/common/utils/test/CMakeFiles/utils_test.dir/OptionSetTest.cpp.o
[38/62] Building CXX object eden/common/utils/test/CMakeFiles/utils_test.dir/MemoryTest.cpp.o
[39/62] Building CXX object eden/common/telemetry/CMakeFiles/edencommon_telemetry.dir/StatsGroup.cpp.o
[40/62] Building CXX object eden/common/utils/test/CMakeFiles/utils_test.dir/StringConvTest.cpp.o
[41/62] Building CXX object eden/common/telemetry/CMakeFiles/edencommon_telemetry.dir/Tracing.cpp.o
[42/62] Building CXX object eden/common/utils/test/CMakeFiles/utils_test.dir/FileDescriptorTest.cpp.o
[43/62] Building CXX object eden/common/utils/test/CMakeFiles/utils_test.dir/ScopedEnvVar.cpp.o
[44/62] Building CXX object eden/common/utils/test/CMakeFiles/utils_test.dir/RefPtrTest.cpp.o
[45/62] Building CXX object eden/common/utils/test/CMakeFiles/utils_test.dir/StringTest.cpp.o
[46/62] Building CXX object eden/common/utils/CMakeFiles/edencommon_utils.dir/FaultInjector.cpp.o
[47/62] Linking CXX shared library eden/common/utils/libedencommon_utils.dylib
[48/62] Building CXX object eden/common/utils/test/CMakeFiles/utils_test.dir/FileUtilsTest.cpp.o
[49/62] Linking CXX executable eden/common/os/test/os_test
[50/62] Building CXX object eden/common/utils/test/CMakeFiles/utils_test.dir/IoFutureTest.cpp.o
[51/62] Linking CXX shared library eden/common/testharness/libedencommon_testharness.dylib
[52/62] Building CXX object eden/common/utils/test/CMakeFiles/utils_test.dir/ThrowTest.cpp.o
[53/62] Linking CXX shared library eden/common/telemetry/libedencommon_telemetry.dylib
[54/62] Building CXX object eden/common/utils/test/CMakeFiles/utils_test.dir/Utf8Test.cpp.o
[55/62] Building CXX object eden/common/utils/test/CMakeFiles/utils_test.dir/ProcessInfoCacheTest.cpp.o
[56/62] Building CXX object eden/common/utils/test/CMakeFiles/utils_test.dir/SpawnedProcessTest.cpp.o
[57/62] Building CXX object eden/common/utils/test/CMakeFiles/utils_test.dir/UserInfoTest.cpp.o
[58/62] Building CXX object eden/common/utils/test/CMakeFiles/utils_test.dir/UnixSocketTest.cpp.o
[59/62] Building CXX object eden/common/utils/test/CMakeFiles/utils_test.dir/PathFuncsTest.cpp.o
[60/62] Building CXX object eden/common/utils/test/CMakeFiles/utils_test.dir/ImmediateFutureTest.cpp.o
[61/62] Linking CXX executable eden/common/utils/test/utils_test
Running phase: checkPhase
@nix { "action": "setPhase", "phase": "checkPhase" }
check flags: -j12 test
[0/2] Re-checking globbed directories...
[0/2] Running tests...
Test project /tmp/nix-build-edencommon-2024.11.18.00.drv-0/source/build
        Start   1: OptionalProcessId.default_constructor_is_empty
        Start   2: OptionalProcessId.conversion_from_zero_is_safe
        Start   3: OptionalProcessId.value_throws_bad_optional_access
        Start   4: OptionalProcessId.can_be_key_in_map
        Start   5: OptionalProcessId.can_be_key_in_unordered_map
        Start   6: OptionalProcessId.ostream_format_empty
        Start   7: OptionalProcessId.ostream_format
        Start   8: OptionalProcessId.fmt_format
        Start   9: ProcessId.negative_process_IDs_are_disallowed_on_unix
        Start  10: ProcessId.zero_complement_is_invalid_process_id
        Start  11: ProcessId.can_be_key_in_map
        Start  12: ProcessId.can_be_key_in_unordered_map
  1/164 Test   #3: OptionalProcessId.value_throws_bad_optional_access ..............................................   Passed    0.03 sec
        Start  13: FileDescriptor.pipeReadWrite
  2/164 Test   #1: OptionalProcessId.default_constructor_is_empty ..................................................   Passed    0.03 sec
  3/164 Test   #2: OptionalProcessId.conversion_from_zero_is_safe ..................................................   Passed    0.03 sec
  4/164 Test   #5: OptionalProcessId.can_be_key_in_unordered_map ...................................................   Passed    0.03 sec
  5/164 Test   #6: OptionalProcessId.ostream_format_empty ..........................................................   Passed    0.03 sec
        Start  14: FileDescriptor.socketPairReadWrite
        Start  15: FileDescriptor.pipeReadvWritev
        Start  16: FileDescriptor.socketPairReadvWritev
        Start  17: FileDescriptor.fileReadvWritev
  6/164 Test   #9: ProcessId.negative_process_IDs_are_disallowed_on_unix ...........................................   Passed    0.03 sec
        Start  18: FileDescriptor.readFullFile
  7/164 Test   #8: OptionalProcessId.fmt_format ....................................................................   Passed    0.03 sec
  8/164 Test  #12: ProcessId.can_be_key_in_unordered_map ...........................................................   Passed    0.03 sec
  9/164 Test  #10: ProcessId.zero_complement_is_invalid_process_id .................................................   Passed    0.03 sec
 10/164 Test   #4: OptionalProcessId.can_be_key_in_map .............................................................   Passed    0.03 sec
 11/164 Test  #11: ProcessId.can_be_key_in_map .....................................................................   Passed    0.03 sec
 12/164 Test   #7: OptionalProcessId.ostream_format ................................................................   Passed    0.03 sec
        Start  19: FileDescriptor.readFullPipe
        Start  20: FileUtilsTest.testWriteReadFile
        Start  21: FileUtilsTest.testReadFileWithNumBytes
        Start  22: FileUtilsTest.testReadPartialFile
        Start  23: FileUtilsTest.testWriteFileAtomicNoTarget
        Start  24: FileUtilsTest.testWriteFileAtomicWithTarget
 13/164 Test  #13: FileDescriptor.pipeReadWrite ....................................................................   Passed    0.03 sec
        Start  25: FileUtilsTest.testWriteFileTruncate
 14/164 Test  #15: FileDescriptor.pipeReadvWritev ..................................................................   Passed    0.03 sec
        Start  26: FileUtilsTest.testGetAllDirectoryEntryNames
 15/164 Test  #16: FileDescriptor.socketPairReadvWritev ............................................................   Passed    0.03 sec
        Start  27: OptionSet.initialization_from_zero
 16/164 Test  #17: FileDescriptor.fileReadvWritev ..................................................................   Passed    0.03 sec
        Start  28: OptionSet.default_is_empty
 17/164 Test  #21: FileUtilsTest.testReadFileWithNumBytes ..........................................................   Passed    0.03 sec
 18/164 Test  #22: FileUtilsTest.testReadPartialFile ...............................................................   Passed    0.03 sec
 19/164 Test  #23: FileUtilsTest.testWriteFileAtomicNoTarget .......................................................   Passed    0.03 sec
        Start  29: OptionSet.assignment_operators
        Start  30: OptionSet.format
        Start  31: OptionSet.containsAllOf
 20/164 Test  #24: FileUtilsTest.testWriteFileAtomicWithTarget .....................................................   Passed    0.03 sec
 21/164 Test  #19: FileDescriptor.readFullPipe .....................................................................   Passed    0.04 sec
 22/164 Test  #18: FileDescriptor.readFullFile .....................................................................   Passed    0.04 sec
        Start  32: OptionSet.intersect
        Start  33: OptionSet.containsAnyOf
        Start  34: OptionSet.containsNoneOf
 23/164 Test  #14: FileDescriptor.socketPairReadWrite ..............................................................   Passed    0.05 sec
        Start  35: ImmediateFuture.get
 24/164 Test  #20: FileUtilsTest.testWriteReadFile .................................................................   Passed    0.04 sec
        Start  36: ImmediateFuture.getTry
 25/164 Test  #27: OptionSet.initialization_from_zero ..............................................................   Passed    0.03 sec
        Start  37: ImmediateFuture.thenValue
 26/164 Test  #28: OptionSet.default_is_empty ......................................................................   Passed    0.02 sec
        Start  38: ImmediateFuture.ensureBasic
 27/164 Test  #30: OptionSet.format ................................................................................   Passed    0.02 sec
        Start  39: ImmediateFuture.ensureThrowInFuture
 28/164 Test  #26: FileUtilsTest.testGetAllDirectoryEntryNames .....................................................   Passed    0.03 sec
        Start  40: ImmediateFuture.ensureThrowInFunc
 29/164 Test  #29: OptionSet.assignment_operators ..................................................................   Passed    0.03 sec
        Start  41: ImmediateFuture.thenValueReturnsImmediateFuture
 30/164 Test  #25: FileUtilsTest.testWriteFileTruncate .............................................................   Passed    0.04 sec
 31/164 Test  #34: OptionSet.containsNoneOf ........................................................................   Passed    0.03 sec
        Start  42: ImmediateFuture.thenTry
        Start  43: ImmediateFuture.exception
 32/164 Test  #31: OptionSet.containsAllOf .........................................................................   Passed    0.03 sec
 33/164 Test  #32: OptionSet.intersect .............................................................................   Passed    0.03 sec
        Start  44: ImmediateFuture.exceptionContinuation
        Start  45: ImmediateFuture.unit
 34/164 Test  #36: ImmediateFuture.getTry ..........................................................................   Passed    0.02 sec
        Start  46: ImmediateFuture.defaultCtor
 35/164 Test  #33: OptionSet.containsAnyOf .........................................................................   Passed    0.03 sec
        Start  47: ImmediateFuture.semi
 36/164 Test  #35: ImmediateFuture.get .............................................................................   Passed    0.03 sec
        Start  48: ImmediateFuture.mutableLambda
 37/164 Test  #37: ImmediateFuture.thenValue .......................................................................   Passed    0.03 sec
        Start  49: ImmediateFuture.getTimeout
 38/164 Test  #39: ImmediateFuture.ensureThrowInFuture .............................................................   Passed    0.03 sec
        Start  50: ImmediateFuture.makeImmediateFutureWith
 39/164 Test  #40: ImmediateFuture.ensureThrowInFunc ...............................................................   Passed    0.03 sec
        Start  51: ImmediateFuture.makeImmediateFutureWithIsEager
 40/164 Test  #38: ImmediateFuture.ensureBasic .....................................................................   Passed    0.03 sec
        Start  52: ImmediateFuture.isReady_from_value
 41/164 Test  #41: ImmediateFuture.thenValueReturnsImmediateFuture .................................................   Passed    0.03 sec
        Start  53: ImmediateFuture.isReady_from_completed_SemiFuture
 42/164 Test  #42: ImmediateFuture.thenTry .........................................................................   Passed    0.03 sec
        Start  54: ImmediateFuture.ready_ImmediateFuture_thenValue_is_also_ready
 43/164 Test  #45: ImmediateFuture.unit ............................................................................   Passed    0.03 sec
 44/164 Test  #48: ImmediateFuture.mutableLambda ...................................................................   Passed    0.03 sec
        Start  55: ImmediateFuture.ImmediateFuture_does_not_run_SemiFuture_callbacks_until_scheduled_on_executor
        Start  56: ImmediateFuture.collectAllImmediate
 45/164 Test  #43: ImmediateFuture.exception .......................................................................   Passed    0.03 sec
        Start  57: ImmediateFuture.collectAllSemi
 46/164 Test  #47: ImmediateFuture.semi ............................................................................   Passed    0.03 sec
        Start  58: ImmediateFuture.collectAllMixed
 47/164 Test  #44: ImmediateFuture.exceptionContinuation ...........................................................   Passed    0.04 sec
        Start  59: ImmediateFuture.collectUncopyable
 48/164 Test  #50: ImmediateFuture.makeImmediateFutureWith .........................................................   Passed    0.02 sec
        Start  60: ImmediateFuture.collectAllOrdering
 49/164 Test  #46: ImmediateFuture.defaultCtor .....................................................................   Passed    0.04 sec
        Start  61: ImmediateFuture.makeImmediateFutureError
 50/164 Test  #51: ImmediateFuture.makeImmediateFutureWithIsEager ..................................................   Passed    0.02 sec
        Start  62: ImmediateFuture.collectAllTuple
 51/164 Test  #49: ImmediateFuture.getTimeout ......................................................................   Passed    0.03 sec
        Start  63: ImmediateFuture.collectAllTupleSemi
 52/164 Test  #52: ImmediateFuture.isReady_from_value ..............................................................   Passed    0.03 sec
        Start  64: ImmediateFuture.collectAllTupleSemiReady
 53/164 Test  #54: ImmediateFuture.ready_ImmediateFuture_thenValue_is_also_ready ...................................   Passed    0.03 sec
        Start  65: ImmediateFuture.collectAllSafeTuple
 54/164 Test  #53: ImmediateFuture.isReady_from_completed_SemiFuture ...............................................   Passed    0.03 sec
        Start  66: ImmediateFuture.collectAllSafeTupleError
 55/164 Test  #58: ImmediateFuture.collectAllMixed .................................................................   Passed    0.02 sec
        Start  67: ImmediateFuture.collectAllSafeTupleValid
 56/164 Test  #55: ImmediateFuture.ImmediateFuture_does_not_run_SemiFuture_callbacks_until_scheduled_on_executor ...   Passed    0.03 sec
        Start  68: ImmediateFuture.collectAllSafeVector
 57/164 Test  #56: ImmediateFuture.collectAllImmediate .............................................................   Passed    0.03 sec
 58/164 Test  #57: ImmediateFuture.collectAllSemi ..................................................................   Passed    0.03 sec
        Start  69: ImmediateFuture.collectAllSafeVectorError
        Start  70: ImmediateFuture.collectAllSafeVectorValid
 59/164 Test  #59: ImmediateFuture.collectUncopyable ...............................................................   Passed    0.03 sec
        Start  71: ImmediateFuture.unit_method
 60/164 Test  #60: ImmediateFuture.collectAllOrdering ..............................................................   Passed    0.03 sec
        Start  72: ImmediateFuture.unit_method_error
 61/164 Test  #63: ImmediateFuture.collectAllTupleSemi .............................................................   Passed    0.03 sec
        Start  73: ImmediateFuture.thenError
 62/164 Test  #61: ImmediateFuture.makeImmediateFutureError ........................................................   Passed    0.03 sec
        Start  74: ImmediateFuture.thenErrorVoid
 63/164 Test  #62: ImmediateFuture.collectAllTuple .................................................................   Passed    0.03 sec
        Start  75: ImmediateFuture.thenErrorSemiValue
 64/164 Test  #64: ImmediateFuture.collectAllTupleSemiReady ........................................................   Passed    0.03 sec
        Start  76: ImmediateFuture.thenErrorSemiError
 65/164 Test  #66: ImmediateFuture.collectAllSafeTupleError ........................................................   Passed    0.03 sec
        Start  77: ImmediateFuture.not_ready
 66/164 Test  #67: ImmediateFuture.collectAllSafeTupleValid ........................................................   Passed    0.03 sec
        Start  78: ImmediateFuture.constructionFromCompatible
 67/164 Test  #65: ImmediateFuture.collectAllSafeTuple .............................................................   Passed    0.03 sec
 68/164 Test  #68: ImmediateFuture.collectAllSafeVector ............................................................   Passed    0.03 sec
        Start  79: ImmediateFuture.destructors_are_called_by_ImmediateFuture_move
        Start  80: ImmediateFuture.invalid_if_makeEmpty
 69/164 Test  #70: ImmediateFuture.collectAllSafeVectorValid .......................................................   Passed    0.03 sec
        Start  81: ImmediateFuture.invalid_if_moved_from
 70/164 Test  #69: ImmediateFuture.collectAllSafeVectorError .......................................................   Passed    0.03 sec
        Start  82: ImmediateFuture.in_place_construction
 71/164 Test  #71: ImmediateFuture.unit_method .....................................................................   Passed    0.03 sec
 72/164 Test  #72: ImmediateFuture.unit_method_error ...............................................................   Passed    0.03 sec
        Start  83: ImmediateFuture.in_place_construction_multiple_arguments
        Start  84: ImmediateFuture.conversion_from_ready_Future
 73/164 Test  #74: ImmediateFuture.thenErrorVoid ...................................................................   Passed    0.03 sec
        Start  85: ImmediateFuture.conversion_from_nonready_Future
 74/164 Test  #73: ImmediateFuture.thenError .......................................................................   Passed    0.03 sec
 75/164 Test  #75: ImmediateFuture.thenErrorSemiValue ..............................................................   Passed    0.03 sec
        Start  86: ImmediateFuture.then_with_Future
        Start  87: ImmediateFuture.QueuedImmediateExecutor_is_unsafe
 76/164 Test  #78: ImmediateFuture.constructionFromCompatible ......................................................   Passed    0.03 sec
        Start  88: IoFuture.read
 77/164 Test  #76: ImmediateFuture.thenErrorSemiError ..............................................................   Passed    0.03 sec
        Start  89: IoFuture.readTimeout
 78/164 Test  #77: ImmediateFuture.not_ready .......................................................................   Passed    0.03 sec
        Start  90: IoFuture.multiRead
 79/164 Test  #79: ImmediateFuture.destructors_are_called_by_ImmediateFuture_move ..................................   Passed    0.03 sec
        Start  91: Memory.StdStringMemoryUsage
 80/164 Test  #81: ImmediateFuture.invalid_if_moved_from ...........................................................   Passed    0.03 sec
        Start  92: Memory.FBStringMemoryUsage
 81/164 Test  #80: ImmediateFuture.invalid_if_makeEmpty ............................................................   Passed    0.03 sec
        Start  93: PathFuncs.Sanity
 82/164 Test  #83: ImmediateFuture.in_place_construction_multiple_arguments ........................................   Passed    0.03 sec
        Start  94: PathFuncs.StringCompare
 83/164 Test  #82: ImmediateFuture.in_place_construction ...........................................................   Passed    0.03 sec
        Start  95: PathFuncs.Iterate
 84/164 Test  #87: ImmediateFuture.QueuedImmediateExecutor_is_unsafe ...............................................   Passed    0.03 sec
        Start  96: PathFuncs.IteratorDecrement
 85/164 Test  #85: ImmediateFuture.conversion_from_nonready_Future .................................................   Passed    0.03 sec
        Start  97: PathFuncs.IterateComponents
 86/164 Test  #86: ImmediateFuture.then_with_Future ................................................................   Passed    0.03 sec
        Start  98: PathFuncs.IterateSuffixes
 87/164 Test  #84: ImmediateFuture.conversion_from_ready_Future ....................................................   Passed    0.04 sec
        Start  99: PathFuncs.InitializeFromIter
 88/164 Test  #88: IoFuture.read ...................................................................................   Passed    0.03 sec
        Start 100: PathFuncs.Hash20
 89/164 Test  #91: Memory.StdStringMemoryUsage .....................................................................   Passed    0.03 sec
        Start 101: PathFuncs.Stream
 90/164 Test  #92: Memory.FBStringMemoryUsage ......................................................................   Passed    0.03 sec
        Start 102: PathFuncs.ImplicitPiece
 91/164 Test  #93: PathFuncs.Sanity ................................................................................   Passed    0.03 sec
        Start 103: PathFuncs.PathComponent
 92/164 Test  #94: PathFuncs.StringCompare .........................................................................   Passed    0.03 sec
        Start 104: PathFuncs.RelativePath
 93/164 Test  #89: IoFuture.readTimeout ............................................................................   Passed    0.04 sec
        Start 105: PathFuncs.AbsolutePath
 94/164 Test  #95: PathFuncs.Iterate ...............................................................................   Passed    0.03 sec
 95/164 Test  #98: PathFuncs.IterateSuffixes .......................................................................   Passed    0.03 sec
        Start 106: PathFuncs.relativize_memory_safety
        Start 107: PathFuncs.dirname
 96/164 Test  #96: PathFuncs.IteratorDecrement .....................................................................   Passed    0.03 sec
 97/164 Test  #97: PathFuncs.IterateComponents .....................................................................   Passed    0.03 sec
        Start 108: PathFuncs.basename
        Start 109: PathFuncs.isSubDir
 98/164 Test  #99: PathFuncs.InitializeFromIter ....................................................................   Passed    0.03 sec
        Start 110: PathFuncs.findParent
 99/164 Test  #90: IoFuture.multiRead ..............................................................................   Passed    0.05 sec
        Start 111: PathFuncs.fmt
100/164 Test #101: PathFuncs.Stream ................................................................................   Passed    0.03 sec
        Start 112: PathFuncs.fmt_const_ref
101/164 Test #102: PathFuncs.ImplicitPiece .........................................................................   Passed    0.03 sec
        Start 113: PathFuncs.splitFirst
102/164 Test #100: PathFuncs.Hash20 ................................................................................   Passed    0.03 sec
        Start 114: PathFuncs.canonicalPath
103/164 Test #103: PathFuncs.PathComponent .........................................................................   Passed    0.03 sec
        Start 115: PathFuncs.joinAndNormalize
104/164 Test #105: PathFuncs.AbsolutePath ..........................................................................   Passed    0.02 sec
        Start 116: PathFuncs.realpath
105/164 Test #104: PathFuncs.RelativePath ..........................................................................   Passed    0.03 sec
        Start 117: PathFuncs.expandUser
106/164 Test #106: PathFuncs.relativize_memory_safety ..............................................................   Passed    0.02 sec
        Start 118: PathFuncs.comparison
107/164 Test #107: PathFuncs.dirname ...............................................................................   Passed    0.03 sec
        Start 119: PathFuncs.comparisonInsensitive
108/164 Test #109: PathFuncs.isSubDir ..............................................................................   Passed    0.02 sec
        Start 120: PathFuncs.localDirCreateRemove
109/164 Test #108: PathFuncs.basename ..............................................................................   Passed    0.03 sec
        Start 121: PathFuncs.noThrow
110/164 Test #112: PathFuncs.fmt_const_ref .........................................................................   Passed    0.02 sec
        Start 122: PathFuncs.HashEquality
111/164 Test #111: PathFuncs.fmt ...................................................................................   Passed    0.03 sec
        Start 123: PathFuncs.move_or_copy
112/164 Test #113: PathFuncs.splitFirst ............................................................................   Passed    0.03 sec
        Start 124: ProcessInfoCache.getProcPidCmdLine
113/164 Test #110: PathFuncs.findParent ............................................................................   Passed    0.04 sec
        Start 125: ProcessInfoCache.readMyPidsName
114/164 Test #115: PathFuncs.joinAndNormalize ......................................................................   Passed    0.02 sec
        Start 126: ProcessInfoCache.expireMyPidsName
115/164 Test #118: PathFuncs.comparison ............................................................................   Passed    0.02 sec
        Start 127: ProcessInfoCache.addFromMultipleThreads
116/164 Test #119: PathFuncs.comparisonInsensitive .................................................................   Passed    0.02 sec
        Start 128: ProcessInfoCache.faultinjector
117/164 Test #114: PathFuncs.canonicalPath .........................................................................   Passed    0.04 sec
        Start 129: ProcessInfoCache.multipleLookups
118/164 Test #116: PathFuncs.realpath ..............................................................................   Passed    0.03 sec
        Start 130: ProcessInfoCache.testSlCommandlineCleaning
119/164 Test #121: PathFuncs.noThrow ...............................................................................   Passed    0.03 sec
        Start 131: ProcessInfoCache.testBuckCommandlineCleaning
120/164 Test #120: PathFuncs.localDirCreateRemove ..................................................................   Passed    0.03 sec
121/164 Test #117: PathFuncs.expandUser ............................................................................   Passed    0.04 sec
        Start 132: Fixture.lookup_expires
        Start 133: RefCounted.make_and_release
122/164 Test #122: PathFuncs.HashEquality ..........................................................................   Passed    0.03 sec
        Start 134: RefCounted.access
123/164 Test #124: ProcessInfoCache.getProcPidCmdLine ..............................................................   Passed    0.02 sec
        Start 135: RefCounted.moving
124/164 Test #125: ProcessInfoCache.readMyPidsName .................................................................   Passed    0.02 sec
        Start 136: RefCounted.copy_is_explicit
125/164 Test #123: PathFuncs.move_or_copy ..........................................................................   Passed    0.03 sec
        Start 137: RefCounted.copies_hold_a_reference
126/164 Test #126: ProcessInfoCache.expireMyPidsName ...............................................................   Passed    0.03 sec
        Start 138: RefCounted.singletons_dont_deallocate
127/164 Test #130: ProcessInfoCache.testSlCommandlineCleaning ......................................................   Passed    0.02 sec
        Start 139: RefCounted.reset
128/164 Test #129: ProcessInfoCache.multipleLookups ................................................................   Passed    0.03 sec
        Start 140: RefCounted.convert_to_base_owned
129/164 Test #128: ProcessInfoCache.faultinjector ..................................................................   Passed    0.03 sec
        Start 141: RefCounted.convert_to_base_singleton
130/164 Test #127: ProcessInfoCache.addFromMultipleThreads .........................................................***Failed    0.03 sec
Running main() from gmock_main.cc
Note: Google Test filter = ProcessInfoCache.addFromMultipleThreads
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from ProcessInfoCache
[ RUN      ] ProcessInfoCache.addFromMultipleThreads
/tmp/nix-build-edencommon-2024.11.18.00.drv-0/source/eden/common/utils/test/ProcessInfoCacheTest.cpp:57: Failure
Expected equality of these values:
  1
  results.size()
    Which is: 0

[  FAILED  ] ProcessInfoCache.addFromMultipleThreads (0 ms)
[----------] 1 test from ProcessInfoCache (0 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (0 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] ProcessInfoCache.addFromMultipleThreads

 1 FAILED TEST

        Start 142: RefCounted.RefPtr_of_Derived_cast_to_RefPtr_of_Base
131/164 Test #134: RefCounted.access ...............................................................................   Passed    0.02 sec
        Start 143: RefCounted.Derived_as_to_second_Base_with_offset
132/164 Test #131: ProcessInfoCache.testBuckCommandlineCleaning ....................................................   Passed    0.03 sec
        Start 144: RefCounted.bumping_reference_count_through_as_affects_parent
133/164 Test #133: RefCounted.make_and_release .....................................................................   Passed    0.03 sec
        Start 145: RefCounted.no_reference_bumping_when_as_on_singleton
134/164 Test #135: RefCounted.moving ...............................................................................   Passed    0.03 sec
        Start 146: SpawnedProcess.cwd_slash
135/164 Test #136: RefCounted.copy_is_explicit .....................................................................   Passed    0.03 sec
        Start 147: SpawnedProcess.cwd_inherit
136/164 Test #132: Fixture.lookup_expires ..........................................................................   Passed    0.03 sec
        Start 148: SpawnedProcess.pipe
137/164 Test #137: RefCounted.copies_hold_a_reference ..............................................................   Passed    0.03 sec
        Start 149: SpawnedProcess.stresstest_pipe_output
138/164 Test #138: RefCounted.singletons_dont_deallocate ...........................................................   Passed    0.03 sec
        Start 150: SpawnedProcess.inputThreaded
139/164 Test #139: RefCounted.reset ................................................................................   Passed    0.03 sec
        Start 151: SpawnedProcess.inputNotThreaded
140/164 Test #140: RefCounted.convert_to_base_owned ................................................................   Passed    0.02 sec
        Start 152: SpawnedProcess.shellQuoting
141/164 Test #142: RefCounted.RefPtr_of_Derived_cast_to_RefPtr_of_Base .............................................   Passed    0.02 sec
        Start 153: String.starts_with
142/164 Test #141: RefCounted.convert_to_base_singleton ............................................................   Passed    0.03 sec
        Start 154: String.ends_with
143/164 Test #144: RefCounted.bumping_reference_count_through_as_affects_parent ....................................   Passed    0.02 sec
        Start 155: ThrowTest.throw__takes_fmt_views
144/164 Test #145: RefCounted.no_reference_bumping_when_as_on_singleton ............................................   Passed    0.02 sec
        Start 156: ThrowTest.throwf_takes_fmt_views
145/164 Test #143: RefCounted.Derived_as_to_second_Base_with_offset ................................................   Passed    0.03 sec
        Start 157: UnixSocket.getRemoteUID
146/164 Test #146: SpawnedProcess.cwd_slash ........................................................................   Passed    0.03 sec
        Start 158: UnixSocket.sendDataAndFiles
147/164 Test #148: SpawnedProcess.pipe .............................................................................   Passed    0.03 sec
        Start 159: FutureUnixSocket.receiveQueue
148/164 Test #147: SpawnedProcess.cwd_inherit ......................................................................   Passed    0.03 sec
        Start 160: FutureUnixSocket.attachEventBase
149/164 Test #150: SpawnedProcess.inputThreaded ....................................................................   Passed    0.03 sec
        Start 161: UserInfo.initFromSudo
150/164 Test #154: String.ends_with ................................................................................   Passed    0.02 sec
        Start 162: UserInfo.lookup
151/164 Test #152: SpawnedProcess.shellQuoting .....................................................................   Passed    0.03 sec
        Start 163: Utf8Test.isValidUtf8
152/164 Test #156: ThrowTest.throwf_takes_fmt_views ................................................................   Passed    0.02 sec
        Start 164: Utf8String.ensureValidUtf8
153/164 Test #153: String.starts_with ..............................................................................   Passed    0.03 sec
154/164 Test #155: ThrowTest.throw__takes_fmt_views ................................................................   Passed    0.03 sec
155/164 Test #151: SpawnedProcess.inputNotThreaded .................................................................   Passed    0.04 sec
156/164 Test #157: UnixSocket.getRemoteUID .........................................................................   Passed    0.03 sec
157/164 Test #159: FutureUnixSocket.receiveQueue ...................................................................   Passed    0.02 sec
158/164 Test #160: FutureUnixSocket.attachEventBase ................................................................   Passed    0.03 sec
159/164 Test #162: UserInfo.lookup .................................................................................   Passed    0.02 sec
160/164 Test #161: UserInfo.initFromSudo ...........................................................................   Passed    0.03 sec
161/164 Test #163: Utf8Test.isValidUtf8 ............................................................................   Passed    0.02 sec
162/164 Test #164: Utf8String.ensureValidUtf8 ......................................................................   Passed    0.02 sec
163/164 Test #158: UnixSocket.sendDataAndFiles .....................................................................   Passed    0.14 sec
164/164 Test #149: SpawnedProcess.stresstest_pipe_output ...........................................................   Passed    6.96 sec

99% tests passed, 1 tests failed out of 164

Total Test time (real) =   7.32 sec

The following tests FAILED:
	127 - ProcessInfoCache.addFromMultipleThreads (Failed)
Errors while running CTest
FAILED: CMakeFiles/test.util 
cd /tmp/nix-build-edencommon-2024.11.18.00.drv-0/source/build && /nix/store/x8qyfphi04lhj621if89f8il33v9dnny-cmake-3.30.5/bin/ctest --force-new-ctest-process
ninja: build stopped: subcommand failed.
```

</details>

## Additional context


## Metadata

```
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.6.0, macOS 14.7.2`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.24.11`
 - nixpkgs: `/nix/store/1jphxmkkqvz69yg0fq6z69dcpfkyy0l9-source`
```

## Notify maintainers

@emilazy
@techknowlogick
@K900

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ed/edencommon/package.nix
+++ b/pkgs/by-name/ed/edencommon/package.nix
@@ -75,6 +75,24 @@ stdenv.mkDerivation (finalAttrs: {
 
   doCheck = true;
 
+  checkPhase = ''
+    runHook preCheck
+
+    # Skip flaky test on darwin
+    ctest -j $NIX_BUILD_CORES --output-on-failure ${
+      lib.optionalString stdenv.hostPlatform.isDarwin (
+        lib.escapeShellArgs [
+          "--exclude-regex"
+          (lib.concatMapStringsSep "|" (test: "^${lib.escapeRegex test}$") [
+            "ProcessInfoCache.addFromMultipleThreads"
+          ])
+        ]
+      )
+    }
+
+    runHook postCheck
+  '';
+
   postPatch = ''
     # The CMake build requires the FBThrift Python support even though
     # it’s not used, presumably because of the relevant code having

